spyder/widgets/helperwidgets.py

```diff
@@ -283,6 +288,84 @@ def paintEvent(self, event):
             self._refresh()
 
 
+# Valid finder chars. To be improved
+VALID_ACCENT_CHARS = "ÁÉÍOÚáéíúóàèìòùÀÈÌÒÙâêîôûÂÊÎÔÛäëïöüÄËÏÖÜñÑ"
+VALID_FINDER_CHARS = r"[A-Za-z\s{0}]".format(VALID_ACCENT_CHARS)
```

Comment:
Please move these constants to the beginning of this file.

Also, how do these constants relate to filtering variables in the Variable Explorer? I say it because in Python 3 any utf-8 character can be a variable, so users are not limited to only using non-ascii variables, as in Python 2.

Reply:
This constants are used as a regex to check the input of the search field, so you can't write something like `_` but you can write something like `á` as a valid search text:

![image](https://user-images.githubusercontent.com/16781833/61499997-d2499a80-a98e-11e9-8760-45e625380dae.png)

Reply:
Ok, that works correctly for the keyboard shortcuts filtering but it doesn't work here for Python 3.

So we need a VALID_VARIABLES constant here and use it to filter variables, but with different values for Python 2 and 3:

* For Python 2 it only needs to ascii characters, digits and the `_` character.
* For Python 3 it needs to contain *any* utf-8 character.

For reference, please see the section called *Variable Names* in

https://realpython.com/python-variables/

You can also see

https://stackoverflow.com/questions/36330860/pythonically-check-if-a-variable-name-is-valid